bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelStateDescriptionProvider.java

```diff
@@ -68,7 +68,10 @@ public StateDescription getStateDescription(String itemName, Locale locale) {
             if (channel != null) {
                 ChannelType channelType = thingTypeRegistry.getChannelType(channel, locale);
                 if (channelType != null) {
-                    StateDescription stateDescription = channelType.getState();
+                    StateDescription stateDescription = thingTypeRegistry.getStateDescription(channel, locale);
```

Comment:
if you use the service reference mentioned above you should be able to query your `DynamicStateDescriptionRegistry` directly and not through the `ThingTypeRegistry`.